### PR TITLE
When is ssh -A insecure?

### PR DESCRIPTION
**Pull request by marpo60**

### CODE REVIEW

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  email: marcelo@wyeworks.com
+  twitter_handle: marpo60
+  github_handle: marpo60
+  image: /images/team/marcelo-dominguez.jpg
+  description: Software engineer - Curious by nature - Sports lover - Tech guy
+---
+
+## Introduction
+
+###What's the authentication agent forwarding for?
```
Comment (spastorino) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: missing white space

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+### How do we use it?
+
+{% codeblock lang:sh %}
+  user@laptop$ ssh -A serverA
+  user@serverA$ ssh serverB
+  user@serverB$
+{% endcodeblock %}
+
+## So, When ssh -A is insecure?
```
Comment (spastorino) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: when without capitalization

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+ - me
+ - Alice (Good)
+ - Eve (Bad)
+
+ Facts:
+
+ * The host machine is our machine where we will create the network
+ topology using virtual machines
+ * The web server is a virtual machine that will emulate a server
+ public accesible.
```
Comment (spastorino) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: emulate a public accessible server?

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+ * The worker server is a virtual machine that will emulate a
+ server accesible only from the web server.
```
Comment (spastorino) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: accessible

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+ First lets add all the keys to the ssh agent of the host
+ machine
+
+ {% codeblock lang:sh %}
+   me@host$ ssh-add alice_id_rsa
+   me@host$ ssh-add eve_id_rsa
+{% endcodeblock %}
+
+ Now we'll open a new terminal and emulate that Alice is working
+ on the worker server
```
Comment (spastorino) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: I don't understand this exactly. As a user am I supposed to do something on this step?.

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 6]
```diff
@@ -0,0 +1,183 @@
+---
+layout: post
+title: When ssh -A is insecure?
```
Comment (smartinez87) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: It should be `When is ssh -A insecure?`

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+
+### How do we use it?
+
+{% codeblock lang:sh %}
+  user@laptop$ ssh -A serverA
+  user@serverA$ ssh serverB
+  user@serverB$
+{% endcodeblock %}
+
+## So, when ssh -A is insecure?
```
Comment (smartinez87) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: `when is ssh -A insecure?`

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+
+The man page has the following note:
+
+> Agent forwarding should be enabled with caution.
+
+So, we have to be carefull with this option, but what are the problems?
+
+With the intention of gaining more insight about the problem. We'll do
+an experiment, we'll set up an scenario that is far form ideal but it's
```
Comment (smartinez87) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: `an scenario` -> `a scenario`

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+   alice@worker:~$
+{% endcodeblock %}
+
+ Now on other terminal lets pretend to be Eve
+
+ {% codeblock lang:sh %}
+   me@host$ ssh -i eve_id_rsa -p 2222 eve@172.17.8.101
+{% endcodeblock %}
+
+ Eve will not use `-A` options because she don't need to use the
```
Comment (smartinez87) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: `don't` -> `doesn't`

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+  github_handle: marpo60
+  image: /images/team/marcelo-dominguez.jpg
+  description: Software engineer - Curious by nature - Sports lover - Tech guy
+---
+
+## Introduction
+
+### What's the authentication agent forwarding for?
+
+Lets start showing what `man ssh` says about it:
```
Comment (eldano) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: Let's *

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+  user@serverB$
+{% endcodeblock %}
+
+## So, when ssh -A is insecure?
+
+The man page has the following note:
+
+> Agent forwarding should be enabled with caution.
+
+So, we have to be carefull with this option, but what are the problems?
```
Comment (santib) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: careful

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+
+With the intention of gaining more insight about the problem. We'll do
+an experiment, we'll set up a scenario that is far form ideal but it's
```
Comment (santib) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: from

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+ * The host machine is our machine where we will create the network
+ topology using virtual machines
+ * The web server is a virtual machine that will emulate a server
+ public accessible.
+ * The worker server is a virtual machine that will emulate a
+ server accessible only from the web server.
+ * The secret server is a virtual machine that will emulate a
+ server accessible only by Alice from the internet.
+
+ Lets start the servers with `vagrant up`.
```
Comment (Axxiss) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: I think here is `let's` (let us)

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+    root@web:/home/eve# ssh -p 2222 alice@172.17.8.110
+    alice@secret:~$
+{% endcodeblock %}
+
+ So, with this simple stepts now Eve is connected to secret
+ machine as Alice.
+
+## Conclusions
+
+ It's not unusual for users to have sudo permissions of shared
```
Comment (Axxiss) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: permissions **on** shared

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown [hunk 15]
```diff
@@ -0,0 +1,183 @@
...
+ environments so you should avoid using
+ -A option if you don't trust that system. One
+ alternative could be to use `ProxyCommand` instead of `-A`
+ http://backreference.org/2010/02/26/jump-in-with-ssh-and-netcat/
```
Comment (Axxiss) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: link markdown missing

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown (hunk 3 above)
Comment (marpo60) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: I change it for "emulate a server accessible from the internet."

File: source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown (hunk 5 above)
Comment (marpo60) [inline source/_posts/2015-07-06-when-ssh-a-is-insecure.markdown]: Instructions are a bit below

File: source/_posts/2015-07-06-when-is-ssh-a-insecure.markdown [hunk 16]
```diff
@@ -0,0 +1,183 @@
...
+  ssh-xeIzefGEYs
+  -rwx--x--x  1 vagrant vagrant  676 Jan 19 21:34 vagrant-shell
+  drwxrwxrwt  2 root    root    4.0K Jan 19 21:34 .X11-unix
+  eve@web:~$
+{% endcodeblock %}
+
+Eve now becomes root and impersonate into Alice.
+
+{% codeblock lang:sh %}
+  eve@web:~$ sudo su
```
Comment (jmbejar) [inline source/_posts/2015-07-06-when-is-ssh-a-insecure.markdown]: What about adding a note saying that the eve's password is `eve123`?
I had to look at the bootstrap script to figure it out.

Comment (marpo60) [inline source/_posts/2015-07-06-when-is-ssh-a-insecure.markdown]: Agree

File: authors.rb [hunk 17]
```diff
@@ -46,5 +46,13 @@
     github_handle: 'spastorino',
     image: 'santiago-pastorino.jpg',
     description: 'WyeWorks Co-Founder & CTO, Ruby on Rails Core Team Member'
+  },
+  marcelo_dominguez: {
+    name: 'Marcelo Dominugez',
```
Comment (spastorino) [inline authors.rb]: typo here

File: source/_posts/2015-07-06-when-is-ssh-a-insecure.markdown [hunk 18]
```diff
@@ -0,0 +1,183 @@
+---
+layout: post
+title: When is ssh -A insecure?
+date: 2015-07-06 15:27:14 -0300
+comments: true
+author:
+  name: Marcelo Dominugez
```
Comment (spastorino) [inline source/_posts/2015-07-06-when-is-ssh-a-insecure.markdown]: typo here

File: source/_posts/2015-07-06-when-is-ssh-a-insecure.markdown [hunk 19]
```diff
@@ -0,0 +1,183 @@
...
+  user@serverB$
+{% endcodeblock %}
+
+## So, when is ssh -A insecure?
+
+The man page has the following note:
+
+> Agent forwarding should be enabled with caution.
+
+So, we have to be careful with this option, but what are the problems?
```
Comment (spastorino) [inline source/_posts/2015-07-06-when-is-ssh-a-insecure.markdown]: `but what the problems are?`

Comment (marpo60) [inline source/_posts/2015-07-06-when-is-ssh-a-insecure.markdown]: I think its fine the way it is.

**Pull request merged**